salt/minion.py

```diff
@@ -1409,12 +1409,18 @@ def pillar_refresh(self, force_refresh=False):
         '''
         Refresh the pillar
         '''
-        self.opts['pillar'] = salt.pillar.get_pillar(
-            self.opts,
-            self.opts['grains'],
-            self.opts['id'],
-            self.opts['environment'],
-        ).compile_pillar()
+        try:
+            self.opts['pillar'] = salt.pillar.get_pillar(
+                self.opts,
+                self.opts['grains'],
+                self.opts['id'],
+                self.opts['environment'],
+            ).compile_pillar()
+        except SaltClientError:
+            # Do not exit if a pillar refresh fails.
+            log.error('Pillar data could not be refreshed. '
+                      'One or more masters may be down!')
+            pass
```

Comment:
Heh, no. I'm being silly there and deciding what I want to do with a PR still open. I'll clean it up before we merge. :]